dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/task/ParallelTaskGroup.java

```diff
@@ -66,15 +68,16 @@ public T call() throws Exception {
   }
 
   @Override
-  @SuppressWarnings(
-      "FutureReturnValueIgnored") // It's an ExecutorManager, which tracks the Future internally.
-  protected void doRun(TaskRunContext context, CSVPrinter printer, Handle handle) throws Exception {
+  protected void doRun(
+      @Nonnull TaskRunContext context, @Nonnull CSVPrinter printer, @Nonnull Handle handle)
+      throws ExecutionException, InterruptedException {
     // Throws ExecutionException if any sub-task threw. However, runChildTask() is nothrow, so that
     // never happens.
     // We safely publish the CSVPrinter to the ExecutorManager.
     try (ExecutorManager executorManager = new ExecutorManager(context.getExecutorService())) {
       for (Task<?> task : getTasks()) {
-        executorManager.submit(new TaskRunner<>(context, task, printer));
+        TaskRunner runner = new TaskRunner(context, task, printer);
+        executorManager.execute(runner::call);
```

Comment:
The only real purpose of TaskRunner was to implement Callable/Runnable. This ::call constructs a new, anonymous Callable/Runnable which can never have a useful toString() and doubles the number of allocations performed by this method.

Reply:
Regarding `toString`: The current version returns something like `TaskRunner@1234`. Should I add a `toString` to the callable being passed here?

Reply:
About the number of allocations - I understand that this means having two allocations per loop iteration instead of one, not including work done inside `execute` or `call` itself. I looked for evidence to support this, but I don't really get anything like an object initialization (e.g. `<init>`) in the results of compiling similar examples.

The `execute` here wants a call site, which we provide. Neither side needs to create a new object wrapping the reference.

Maybe this worked differently before Java 8?